### ISSUE TITLE
Code seems not working

### ISSUE THREAD

**Issue by canatojr** [limsko/moodle-filter_slider#1]
Hello!

I installed both slider block and filter and activated the slider filter in Site Administration.
The block is working good but I couldn't put filter to work.
So where I have to insert the code [SLIDER-XXXXXX]? I tried in labels of the same course where block was created and in labels of a different course, but not worked.

Best regards!

**Comment by limsko** [comment]
Hi, 

Does the filter is enabled at filters page? moodle-url/admin/filters.php


-- 
Pozdrawiam
Kamil Łuczak
tel. +48 790 551 397
https://t.me/limsko 

W dniu 2020-03-08 11:32, canatojr napisał(a):

> Hello! 
> 
> I installed both slider block and filter and activated the slider filter in Site Administration.
> The block is working good but I couldn't put filter to work.
> So where I have to insert the code [SLIDER-XXXXXX]? I tried in labels of the same course where block was created and in labels of a different course, but not worked. 
> 
> Best regards! 
> 
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub [1], or unsubscribe [2].
 

Links:
------
[1]
https://github.com/limsko/moodle-filter_slider/issues/1?email_source=notifications&amp;email_token=ADC24IS43CLF5R6HGCS5E53RGNX4RA5CNFSM4LDYFPQ2YY3PNVWWK3TUL52HS4DFUVEXG43VMWVGG33NNVSW45C7NFSM4ITLUK7A
[2]

**Comment by canatojr** [comment]
Yes! See the images: [https://drive.google.com/drive/folders/1FQuxNAuJxHYfRWMXLsQojS4p8YwpQ0_Y?usp=sharing]

**Comment by limsko** [comment]
Hi, can You update the plugin and check if error still exists?

**Comment by rizole** [comment]
We have the same issue. We're on 3.5.8+ and the filter isn't working for us on a fresh install of the latest block and filter.
I have tried it on a local 3.8 test moodle and it's working there no problem. I'm going to try a 3.6 upgrade on a live site and see if that makes a difference. The filter is enabled in admin by the way.

**Comment by limsko** [comment]
Hi thanks for sharing I will look into this problem today and try to fix it asap.

Pozdrawiam
Kamil Łuczak
tel. 790 551 397
https://limsko.pl 15 maj 2020 17:12:46 rizole <notifications@github.com>:

**Comment by limsko** [comment]
@rizole Clean install of Moodle 3.5, check this out:
https://moodle.limsko.pl/moodle/35/
Did you enter Block Settings and choose Slider JS library? 
I think the slide works fine, but not necessarily intuitive.
When you add a slide you have to configure it, because by default all navigation elements and automatic slide scrolling are disabled as well.
![configure_slider](https://user-images.githubusercontent.com/12955170/82074971-13076280-96cb-11ea-8ff5-d5e3c075289d.jpg)
![configure_slider_2](https://user-images.githubusercontent.com/12955170/82074982-14d12600-96cb-11ea-8216-3c5ed2124790.jpg)

**Comment by rizole** [comment]
Yeah, I got it working on a clean 3.5 install no problem. We've also upgraded to the latest 3.5.12+ on our production site but the filter still isn't working.
I've tried different themes, not sure what to try next though. Any suggestions welcome.
![image](https://user-images.githubusercontent.com/65400192/82576174-65d89280-9b81-11ea-9f54-24cc4010cb88.png)

**Comment by sgodbehere01** [comment]
Hi there, 

I've been helping rizole with this issue we're seeing above. I've created a clone of the Moodle having issues to experiment on and found the following which you might have a better insight for what's going wrong here: 

If I restore a course from another production Moodle site on the same server (same environment) where this works and restore it to the cloned one, the block appears as it should but where the the slider should appear the expected error appears: "Slider Block with ID 3692 doesn't exists!" because as the course is restored the ID of the accompanying block changes. 

If I manually change the ID in the database (mdl_block_instances) to match what the filter wants (for example 3692 in the above case) the slider proceeds to work. However this only works for a restored course if I change the ID of a pre-existing block. Any attempt to create a new block with the filter as a label doesn't work and produces the issue where the label simply appear as "[SLIDER-xxxxxx]". 

So it seems as though on the cloned non-working site the block and filter are partially working, but for some reason new blocks/filter codes are not being picked up to create the connection between the course content and block. 

Let me know if any of that needs clarifying or if you'd like me to try anything else in regards to this. I've replicated the filters list to copy the Moodle which is working, purged caches etc. 

Kind regards,

Sam
